### PR TITLE
ROX-20751: Add Konflux build pipeline for scanner-db

### PR DESCRIPTION
**Pull request by kylape**
Similar to https://github.com/stackrox/scanner/pull/1334, I'm trying to get Konflux to build scanner images.  This one is for scanner-db.

----

Added by @msugakov :

## Testing done

* I **did not** test the resulting container in **E2E** tests deferring it to the moment we'll be testing the entire product (there's no JIRA ticket for this yet but it's in Miro and I'll transfer it).
* I did a shallow check of local docker-running upstream-built image and Konflux-built image to see them both fail the same way (see below).

```bash
$ podman run --rm -it --entrypoint=/bin/bash -u 0:0 quay.io/stackrox-io/scanner-db:4.3.2
# or
$ podman run --rm -it -e POSTGRES_PASSWORD=password quay.io/redhat-user-workloads/rh-acs-tenant/acs/scanner-db:on-pr-8bdc93becb099f1630a1c0ef5f5689b201f31232
```

<details>
<summary>Produced output</summary>

```
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "C".
The default database encoding has accordingly been set to "SQL_ASCII".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /var/lib/postgresql/data/pgdata ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 100
selecting default shared_buffers ... 128MB
selecting default time zone ... Etc/UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
syncing data to disk ... ok

initdb: warning: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.

Success. You can now start the database server using:

    pg_ctl -D /var/lib/postgresql/data/pgdata -l logfile start

waiting for server to start....2024-02-02 17:00:17.003 UTC [26] FATAL:  could not load server certificate file "/run/secrets/stackrox.io/certs/server.crt": No such file or directory
2024-02-02 17:00:17.003 UTC [26] LOG:  database system is shut down
 stopped waiting
pg_ctl: could not start server
Examine the log output.

```

</details>

### CODE REVIEW

**Comment by msugakov** [comment]
Enterprise Contract before
image

Hoping to see some improvements after.

**Comment by msugakov** [comment]
Enterprise Contract now
image

**Review by msugakov — APPROVED** [review]

**Review by tommartensen — APPROVED** [review]
LGTM

File: image/db/rhel/konflux.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    # We must set version label to prevent inheriting value set in the base stage.
+    # TODO(ROX-20236): configure injection of dynamic version value when it becomes possible.
+    version="0.0.1-todo"
+
+USER root
+
+COPY image/db/pg_hba.conf image/db/postgresql.conf /etc/
+COPY --chown=postgres:postgres image/db/rhel/scripts/docker-entrypoint.sh /usr/local/bin/
+
+RUN dnf upgrade -y --nobest && \
```
Comment (kylape) [inline image/db/rhel/konflux.Dockerfile:24]: this `dnf ugprade` isn't present downstream. I wonder if we can safely remove it.

File: image/db/rhel/konflux.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+COPY blob-pg-definitions.sql.gz /docker-entrypoint-initdb.d/definitions.sql.gz
+
+ENV PG_MAJOR=12 \
+    PGDATA="/var/lib/postgresql/data/pgdata"
+
+ENTRYPOINT ["docker-entrypoint.sh"]
+
+EXPOSE 5432
+CMD ["postgres", "-c", "config_file=/etc/postgresql.conf"]
```
Comment (kylape) [inline image/db/rhel/konflux.Dockerfile:44]: these extra arguments are not present downstream.  I wonder if we can safely remove them.

File: image/db/rhel/konflux.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    source-location="https://github.com/stackrox/scanner" \
+    summary="Scanner DB for RHACS" \
+    url="https://catalog.redhat.com/software/container-stacks/detail/60eefc88ee05ae7c5b8f041c" \
+    # We must set version label to prevent inheriting value set in the base stage.
+    # TODO(ROX-20236): configure injection of dynamic version value when it becomes possible.
+    version="0.0.1-todo"
+
+USER root
+
+COPY image/db/pg_hba.conf image/db/postgresql.conf /etc/
```
Comment (kylape) [inline image/db/rhel/konflux.Dockerfile:21]: breaking these `COPY` commands into multiple lines would match the style of the downstream images (and likely the other scanner dockerfile)

File: image/db/rhel/konflux.Dockerfile (hunk 1 above)
Comment (msugakov) [reply image/db/rhel/konflux.Dockerfile:24]: Well, it's there. See <https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-scanner-db/Dockerfile.in?ref_type=heads#L28>. We do this in all containers in order not to have vulnerabilities in containers: there's sometimes few weeks delay between RPMs are updated and corresponding base containers are released. By updating RPMs, we decrease the chance to go out with CVEs.

File: image/db/rhel/konflux.Dockerfile (hunk 2 above)
Comment (msugakov) [reply image/db/rhel/konflux.Dockerfile:44]: Could it be that we looked at different files or branches? https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-scanner-db/Dockerfile.in?ref_type=heads#L59

File: image/db/rhel/konflux.Dockerfile (hunk 3 above)
Comment (msugakov) [reply image/db/rhel/konflux.Dockerfile]: Multilined them. Feel free to commit on top if you meant not what I made!

**Review by kylape — COMMENTED** [review]
Everything LGTM.  I will note that the e2e tests are failing, though it's likely due to some vuln data being updated causing a mismatch in e2e tests.

**Pull request merged**